flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1356,6 +1361,61 @@ IntrinsicLibrary::genPresent(mlir::Type,
                                           fir::getBase(args[0]));
 }
 
+// SCAN 
+fir::ExtendedValue
+IntrinsicLibrary::genScan(mlir::Type resultType,
+                          llvm::ArrayRef<fir::ExtendedValue> args) {
+
+  assert(args.size() >= 2);
+
+  // required string argument
+  auto string = builder.createBox(loc, args[0]);
+  // required set argument
+  auto set = builder.createBox(loc, args[1]);
+
+  // Handle optional argument, kind
+  // TBD: Where to get default integer kind? For now, use 4.
```

Comment:
`builder.getKindMap().defaultIntegerKind()`

All things related to default kinds in lowering are accessible in `fir::KindMapping` that you can get from the builder.